05-Introducing-Rigid-Body-Molecular-Dynamics/03-Thermodynamic-Quantities-of-Rigid-Bodies.ipynb

```diff
@@ -0,0 +1,354 @@
+    "lj.params[('rigid_body', 'constituent_sphere')] = dict(epsilon=0, sigma=sigma)\n",
+    "lj.params[('constituent_sphere', 'constituent_sphere')] = dict(epsilon=epsilon, sigma=sigma)\n",
+    "lj.r_cut[('constituent_sphere', 'constituent_sphere')] = 4 * sigma \n",
+    "lj.r_cut[('rigid_body', 'rigid_body')] = 0.\n",
+    "lj.r_cut[('rigid_body', 'constituent_sphere')] = 0.\n",
+    "integrator.forces.append(lj)\n",
+    "\n",
+    "rigid_centers_and_free_filter = hoomd.filter.Rigid((\"center\", \"free\"))\n",
+    "nvt = hoomd.md.methods.NVT(kT = temperature, tau = 1., filter = rigid_centers_and_free_filter)\n",
+    "integrator.methods.append(nvt)\n",
+    "\n",
+    "rigid = constrain.Rigid()\n",
+    "constituent_types = ['constituent_sphere','constituent_sphere']\n",
+    "rigid.body['rigid_body'] = {\n",
+    "    \"constituent_types\": constituent_types,\n",
+    "    \"positions\":local_position,\n",
+    "    \"orientations\":[(1,0,0,0)] * len(constituent_types),\n",
+    "    \"charges\":[0.] * len(constituent_types),\n",
+    "    \"diameters\":[2.] * len(constituent_types)}"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "fc46a9ba-34ee-4750-9a5c-be492178c155",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "rigid.create_bodies(sim.state)\n",
+    "integrator.rigid = rigid\n",
+    "sim.operations.integrator = integrator"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "7aa6ec08-36c3-4ea5-b268-4fc710b51e9a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Gsd writer for simulations\n",
+    "logger = hoomd.logging.Logger(['particle','constraint'])\n",
+    "gsd_oper = hoomd.write.GSD(trigger=hoomd.trigger.Periodic(1000), filename='randomization.gsd',log=logger,mode='wb',dynamic=['momentum','property'])\n",
+    "sim.operations += gsd_oper"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fdbd2d72-edf3-4e75-a796-6f3e77abdbf2",
+   "metadata": {},
+   "source": [
+    "## We now want to compute the thermodynamic properties:\n",
+    "\n",
+    "For this example we want to compare the system wide translational kinetic energy calculated by hoomd matches to the translational kinetic energy calculated from the velocities and masses of the rigid bodies. \n",
+    "\n",
+    "We will thereby demonstrate the the thermodynamic quantities of the system are determined by the properties of the rigid bodies (mass, moment of inertia, ect.), and the forces between the constituent particles but the masses of the constituent particles. The constituent particles only define the force field between rigid bodies. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "a59847ce-0581-4f7a-8eb8-61a9d29a4bfe",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "thermodynamic_properties = hoomd.md.compute.ThermodynamicQuantities(\n",
+    "    filter=rigid_centers_and_free_filter)\n",
```

Comment:
Note that you have to pass the correct filter. The way you word the tutorial, it seems that this should happen automatically. This only happens because you exclude the constituent particles from integration and the thermodynamic compute (while including them is an error and results in incorrect behavior). We should emphasize that to properly compute thermodynamic quantities or properly integrate we have to do it this way.

Reply:
ThermodynamicQuantities is designed to work with the All filter in most cases. It knows to ignore constituent particles.